mosaic/mosaic.go

```diff
@@ -0,0 +1,423 @@
+)
+
+// Blocks definition.
+var (
+	HalfBlocks = []Block{
+		{Char: '▀', Coverage: [4]bool{true, true, false, false}, CoverageMap: "██\n  "},   // Upper half block.
+		{Char: '▄', Coverage: [4]bool{false, false, true, true}, CoverageMap: "  \n██"},   // Lower half block.
+		{Char: ' ', Coverage: [4]bool{false, false, false, false}, CoverageMap: "  \n  "}, // Space.
+		{Char: '█', Coverage: [4]bool{true, true, true, true}, CoverageMap: "██\n██"},     // Full block.
+	}
+	QuarterBlocks = []Block{
+		{Char: '▘', Coverage: [4]bool{true, false, false, false}, CoverageMap: "█ \n  "}, // Quadrant upper left.
+		{Char: '▝', Coverage: [4]bool{false, true, false, false}, CoverageMap: " █\n  "}, // Quadrant upper right.
+		{Char: '▖', Coverage: [4]bool{false, false, true, false}, CoverageMap: "  \n█ "}, // Quadrant lower left.
+		{Char: '▗', Coverage: [4]bool{false, false, false, true}, CoverageMap: "  \n █"}, // Quadrant lower right.
+		{Char: '▌', Coverage: [4]bool{true, false, true, false}, CoverageMap: "█ \n█ "},  // Left half block.
+		{Char: '▐', Coverage: [4]bool{false, true, false, true}, CoverageMap: " █\n █"},  // Right half block.
+		{Char: '▀', Coverage: [4]bool{true, true, false, false}, CoverageMap: "██\n  "},  // Upper half block (already added).
+		{Char: '▄', Coverage: [4]bool{false, false, true, true}, CoverageMap: "  \n██"},  // Lower half block (already added).
+	}
+	ComplexBlocks = []Block{
+		{Char: '▙', Coverage: [4]bool{true, false, true, true}, CoverageMap: "█ \n██"},  // Quadrant upper left and lower half.
+		{Char: '▟', Coverage: [4]bool{false, true, true, true}, CoverageMap: " █\n██"},  // Quadrant upper right and lower half.
+		{Char: '▛', Coverage: [4]bool{true, true, true, false}, CoverageMap: "██\n█ "},  // Quadrant upper half and lower left.
+		{Char: '▜', Coverage: [4]bool{true, true, false, true}, CoverageMap: "██\n █"},  // Quadrant upper half and lower right.
+		{Char: '▚', Coverage: [4]bool{true, false, false, true}, CoverageMap: "█ \n █"}, // Quadrant upper left and lower right.
+		{Char: '▞', Coverage: [4]bool{false, true, true, false}, CoverageMap: " █\n█ "}, // Quadrant upper right and lower left.
+	}
+)
+
+// Block represents different Unicode block characters.
+type Block struct {
+	Char        rune
+	Coverage    [4]bool // Which parts of the block are filled (true = filled).
+	CoverageMap string  // Visual representation of coverage for debugging.
+}
+
+// Symbol represents the symbol type to use when rendering the image.
+type Symbol string
+
+// Symbol types.
+const (
+	AllSymbols     Symbol = "all"
+	HalfSymbols    Symbol = "half"
+	QuarterSymbols Symbol = "quarter"
+)
+
+// Scale represents scale mode that will be used when rendering the image.
+type Scale uint8
+
+// Symbol types.
+const (
+	None Scale = iota
+	Fit
+	Stretch
+	Center
+)
+
+// Render mosaic with default values
+func Render(img image.Image) string {
+	m := New()
+	return m.Render(img)
+}
+
+// Options contains all configurable settings.
```

Comment:
```suggestion
// Mosaic is a renderer and its options.
```

also, maybe this should be called Renderer? so usage would be `mosaic.New() // returns a Renderer`